nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/CountText.java

```diff
@@ -0,0 +1,327 @@
+
+    // Attribute keys
+    public static final String TEXT_LINE_COUNT = "text.line.count";
+    public static final String TEXT_LINE_NONEMPTY_COUNT = "text.line.nonempty.count";
+    public static final String TEXT_WORD_COUNT = "text.word.count";
+    public static final String TEXT_CHARACTER_COUNT = "text.character.count";
+
+
+    public static final PropertyDescriptor TEXT_LINE_COUNT_PD = new PropertyDescriptor.Builder()
+            .name("text-line-count")
+            .displayName("Count Lines")
+            .description("If enabled, will count the number of lines present in the incoming text.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor TEXT_LINE_NONEMPTY_COUNT_PD = new PropertyDescriptor.Builder()
+            .name("text-line-nonempty-count")
+            .displayName("Count Non-Empty Lines")
+            .description("If enabled, will count the number of lines that contain a non-whitespace character present in the incoming text.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor TEXT_WORD_COUNT_PD = new PropertyDescriptor.Builder()
+            .name("text-word-count")
+            .displayName("Count Words")
+            .description("If enabled, will count the number of words (alphanumeric character groups bounded by whitespace)" +
+                    " present in the incoming text. Common logical delimiters [_-.] do not bound a word unless 'Split Words on Symbols' is true.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor TEXT_CHARACTER_COUNT_PD = new PropertyDescriptor.Builder()
+            .name("text-character-count")
+            .displayName("Count Characters")
+            .description("If enabled, will count the number of characters (including whitespace and symbols, but not including newlines and carriage returns) present in the incoming text.")
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor SPLIT_WORDS_ON_SYMBOLS_PD = new PropertyDescriptor.Builder()
+            .name("split-words-on-symbols")
+            .displayName("Split Words on Symbols")
+            .description("If enabled, the word count will identify strings separated by common logical delimiters [_-.] as independent words (ex. split-words-on-symbols = 4 words).")
```

Comment:
This is minor issue, but I noticed when rendered using variable-width font in the tooltip the character group `[_-.]` is difficult to read (see screenshot). It might be better to write this a different way for legibility.

![image](https://user-images.githubusercontent.com/5102332/34689958-548c1bbe-f485-11e7-9fc3-a32f7776c1ef.png)